### DISCUSSION TITLE
Creating a hollow cylinder geometry

### DISCUSSION THREAD

**Discussion by Tbuttts21** [firemodels/fds#12286] [General]
I've been reading through the FDS manual and am trying to determine a good way (if one exists) to create a hollow cylinder geometry. I know that you can create cylinders using &GEOM, but I want it to be hollow instead of solid. 

Is there a good way of doing this? Or should I just attempt to create it manually?

**Comment by marcosvanella** [comment]
You can make it using preprocessors, either BlenderFDS or Pyrosim.

**Reply by Tbuttts21** [reply]
Yes, that is true. I was hoping there was a way to do it in FDS itself so I could automate a process in Python.

**Reply by marcosvanella** [reply]
Then I'd say you need to build the triangulation in python using your hollow cylinder parameters. We don't have a hollow cylinder feature currently. Another option (haven't tried it) would be using an extruded polygon with the cross section shape of the tube as a closed "C", but that might be just as costly on your side.

**Comment by emanuelegissi** [comment]
BlenderFDS is written in Python ;-)

Il ven 15 dic 2023, 18:44 marcosvanella ***@***.***> ha
scritto:

**Reply by Koranten** [reply]
Can I write an .STL file with a CAD software other than BlenderFDS, and read this STL file with FDS? I'm trying to do this hollow cylinder stuff.

**Comment by rmcdermo** [comment]
If you are going to be using FDS with complex geometry, I highly recommend you either learn BlenderFDS or Pyrosim.  As someone who had to learn Blender and BlenderFDS, I admit it is a heavy lift.  But there are numerous youtube videos for helping you.  And, as Emanuele points out, it is scriptable, though I have not yet even gotten that far.  I have found that the new MOVE feature in FDS means I can just make geometries in a simple way in Blender and then export to the FDS format and MOVE (also does reorientation) however I need to.  Creating a hollow cylinder in BlenderFDS is not hard once you know have a basic handle on Blender.  Again, not saying this is trivial, only that you are better off investing your time learning this than in writing a one off script for generating the FDS geometry.  No, you cannot just import the STL into FDS.  You'd first have to get it into Blender anyway, so I find it much easier usually to just build the geometry in Blender.  My two cents...

**Reply by Koranten** [reply]
Ok, thanks. I'm not going to do complex geometries, it's really just gonna be a cylinder room. I already used BlenderFDS in the past to create the blocks of a cylinder room, but it was blocky by definition years ago. Now I'd like to use the new beta GEOM stuff to have a smooth hollow cylinder, so I guess I'm going to BlenderFDS again.

**Comment by aleclementi** [comment]
Hi @Tbuttts21  ,

I had the same issue while trying creating a hollow horizontal cylindrical vessel. Eventually I used a script in blender fds. 
[vessel5m.zip](https://github.com/user-attachments/files/20794538/vessel5m.zip)

GEOM makes it smoother, however has limitations. In my case for example i needed to spray water over the cylinder, but the way FDS treats the drops at equatorial line (they fall straight down) did not serve my case. So I had to use the "blocky" approach.
